backend/compact-connect/lambdas/purchases/tests/unit/test_purchase_client.py

```diff
@@ -394,3 +409,148 @@ def test_purchase_client_raises_failed_transaction_exception_when_void_transacti
+            mock_create_transaction_controller=mock_get_merchant_details_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        result = test_purchase_client.validate_and_store_credentials(
+            compact_name='aslp', credentials=self._generate_test_credentials_object()
+        )
+
+        self.assertEqual({'message': 'Successfully verified credentials'}, result)
+
+        call_args = mock_get_merchant_details_controller.call_args.args
+        api_contract_v1_obj = call_args[0]
+        # authentication fields
+        self.assertEqual(MOCK_LOGIN_ID, api_contract_v1_obj.merchantAuthentication.name)
+        self.assertEqual(MOCK_TRANSACTION_KEY, api_contract_v1_obj.merchantAuthentication.transactionKey)
+
+    @patch('purchase_client.getMerchantDetailsController')
+    def test_purchase_client_creates_secret_when_secret_does_not_exist(self, mock_get_merchant_details_controller):
+        from purchase_client import PurchaseClient
+
+        mock_secrets_manager_client = self._generate_mock_secrets_manager_client()
+        self._when_authorize_dot_net_credentials_are_valid(
+            mock_create_transaction_controller=mock_get_merchant_details_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        result = test_purchase_client.validate_and_store_credentials(
+            compact_name='octp', credentials=self._generate_test_credentials_object()
+        )
+
+        self.assertEqual({'message': 'Successfully verified credentials'}, result)
+
+        mock_secrets_manager_client.create_secret.assert_called_once_with(
+            Name='compact-connect/env/test/compact/octp/credentials/payment-processor',
+            SecretString=json.dumps(
+                {
+                    'processor': 'authorize.net',
+                    'api_login_id': MOCK_LOGIN_ID,
+                    'transaction_key': MOCK_TRANSACTION_KEY,
+                }
+            )
+        )
+
+    @patch('purchase_client.getMerchantDetailsController')
+    def test_purchase_client_updates_secret_when_secret_exists(self, mock_get_merchant_details_controller):
+        from purchase_client import PurchaseClient
+
+        mock_secrets_manager_client = self._generate_mock_secrets_manager_client()
+        self._when_authorize_dot_net_credentials_are_valid(
+            mock_create_transaction_controller=mock_get_merchant_details_controller
+        )
+
+        test_purchase_client = PurchaseClient(secrets_manager_client=mock_secrets_manager_client)
+
+        result = test_purchase_client.validate_and_store_credentials(
+            compact_name='aslp', credentials=self._generate_test_credentials_object()
+        )
+
+        self.assertEqual({'message': 'Successfully verified credentials'}, result)
+
+        mock_secrets_manager_client.put_secret_value.assert_called_once_with(
```

Comment:
is this relying on the previous test having saved the secret?

Reply:
No, the mock is set up in this specific test to return a value, which causes the 'put_secret_value' api call to be made rather than the 'create_secret' api call

Reply:
None of these tests are dependent on the ordering of the others

Reply:
Ah, I see. Can you add a brief comment, perhaps on line 503 `compact_name='aslp'`, mentioning that the mock secrets manager has a secret for ASLP? I didn't quite understand this until I went back to the generating function.

Reply:
I've added the comment.